### PR TITLE
OPS: Update to Nav 6

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4e322wxtuwctosbfbn66mzyfvm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11049)

**Comment by Overtorment** [comment]
some tests failed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wxbddlmit2ahdwzixgsgkg53gi?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11056)

**Comment by marcosrdz** [comment]
@limpbrains @Overtorment  i ts-ignored useNavigation in a commit so it passes CI. Either of you know what changes are needed to avoid using ts-ignore?

**Comment by limpbrains** [comment]
@marcosrdz I believe there is not easy fix. You need to type [whole Navigation](https://reactnavigation.org/docs/typescript/). I tried to type one screen, but failed :)
This is definetly should be a separate task to do.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ootnptcby664fon4qgjn5xsyam?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11163)

**Comment by Overtorment** [comment]
e2e still fails?

**Comment by Overtorment** [comment]
@marcosrdz ^^^

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qwjqscunsmpxyrk2h7vycnqp2m?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11535)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/isx3xqhr24yv7vevjw4ahlzscu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11536)

**Comment by limpbrains** [comment]
Can't run detox locally on this branch.
Getting 
```
    The app has crashed, see the details below:

    @Thread detox.primary(146):
    java.lang.RuntimeException: java.util.concurrent.ExecutionException: androidx.test.espresso.IdlingResourceTimeoutException: Wait for [com.wix.detox.reactnative.idlingresources.NetworkIdlingResource] to become idle timed out
    	at androidx.test.espresso.Espresso.onIdle(Espresso.java:15)
    	at com.wix.detox.TestEngineFacade.awaitIdle(TestEngineFacade.kt:12)
    	at com.wix.detox.adapters.server.ReadyActionHandler.handle(DetoxActionHandlers.kt:25)
    	at com.wix.detox.adapters.server.ActionsExecutor.executeAction$lambda-2$lambda-1(DetoxActionsDispatcher.kt:64)
    	at com.wix.detox.adapters.server.ActionsExecutor.$r8$lambda$nk3cF8ntnfFqbJOcVqd04vjb5nY(Unknown Source:0)
    	at com.wix.detox.adapters.server.ActionsExecutor$$ExternalSyntheticLambda0.run(Unknown Source:8)
    	at android.os.Handler.handleCallback(Handler.java:942)
    	at android.os.Handler.dispatchMessage(Handler.java:99)
    	at android.os.Looper.loopOnce(Looper.java:201)
    	at android.os.Looper.loop(Looper.java:288)
    	at com.wix.detox.adapters.server.ActionsExecutor._init_$lambda-0(DetoxActionsDispatcher.kt:50)
    	at com.wix.detox.adapters.server.ActionsExecutor.$r8$lambda$Hi5vyewqQiQqk57AkJy_cHqNGKY(Unknown Source:0)
    	at com.wix.detox.adapters.server.ActionsExecutor$$ExternalSyntheticLambda1.run(Unknown Source:4)
    	at java.lang.Thread.run(Thread.java:1012)
    Caused by: java.util.concurrent.ExecutionException: androidx.test.espresso.IdlingResourceTimeoutException: Wait for [com.wix.detox.reactnative.idlingresources.NetworkIdlingResource] to become idle timed out
    	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
    	at java.util.concurrent.FutureTask.get(FutureTask.java:191)
    	at androidx.test.espresso.Espresso.onIdle(Espresso.java:10
```

Strange
Maybe RN 72 will help

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/esogfnvgviyhzm7yksvkljiiwm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11564)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/sle3wh3pn7pveybf5jqjgld74m?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11613)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bjgnxmpf25xwfe4rod23xdp364?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11614)

**Comment by marcosrdz** [comment]
@limpbrains  some screens were coded "incorrectly" and need to have their params modified to add "merge: true"

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/l6t7tvrcabsqfcmseykpfhlymq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11615)

**Comment by limpbrains** [comment]
Do you mean that every screen should be written assuming that ﻿route.params could be undefined?
I kinda don't like it

**Comment by marcosrdz** [comment]
> Do you mean that every screen should be written assuming that ﻿route.params could be undefined?
> 
> I kinda don't like it 

https://stackoverflow.com/questions/67327642/what-does-merge-do-in-navigation-navigate-react-native

**Comment by limpbrains** [comment]
Got it. I'm thinking, maybe, for qrcode scanner we should try to always use goBack()? 
Also, do you know any other screen that needs to be fixed?

**Comment by marcosrdz** [comment]
> Got it. I'm thinking, maybe, for qrcode scanner we should try to always use goBack()? 
> Also, do you know any other screen that needs to be fixed? 

ScanQRCode onBarScanned should do a navigate to the sender screen with updated params along with the merge:true.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dcd6deqsncaoko6nvpvmkah6bq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11802)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/f2jbexujjvosfqt327dalkyeye?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11803)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rmxae6xgglbimxi4hpntwkbo2u?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11804)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6upuwedfafefxe2zttixrmgbqu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11850)

File: package.json [hunk 1]
```diff
@@ -158,7 +159,7 @@
     "react-native-handoff": "https://github.com/BlueWallet/react-native-handoff#31d005f93d31099d0e564590a3bbd052b8a02b39",
     "react-native-haptic-feedback": "2.2.0",
     "react-native-idle-timer": "https://github.com/BlueWallet/react-native-idle-timer#8587876d68ab5920e79619726aeca9e672beaf2b",
-    "react-native-image-picker": "4.8.5",
+    "react-native-image-picker": "5.2.0",
```
Comment (marcosrdz) [inline package.json:162]: Older version wouldn't compile with any sdk past 32

File: scripts/react-native-camera-kit.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- ../node_modules/react-native-camera-kit/android/src/main/java/com/rncamerakit/CKCamera.kt
```
Comment (limpbrains) [inline scripts/react-native-camera-kit.patch:1]: I don't think patch is needed with latest version of camera kit

File: screen/wallets/pleaseBackup.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ const PleaseBackup: React.FC = () => {
 
   const handleBackButton = useCallback(() => {
     // @ts-ignore: Ignore
-    navigation.dangerouslyGetParent()?.pop();
+    navigation.getParent()?.pop();
```
Comment (limpbrains) [inline screen/wallets/pleaseBackup.tsx:38]: ? Might not be needed

File: android/build.gradle [hunk 4]
```diff
@@ -14,16 +14,15 @@ buildscript {
 
         // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
         ndkVersion = "23.1.7779620"
-        kotlin_version = '1.9.20'
-        kotlinVersion = '1.8.0'
-
+        kotlin_version = '1.7.21'
```
Comment (limpbrains) [inline android/build.gradle:17]: Why are decreasing kotlin version?

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: scripts/react-native-camera-kit.patch (hunk 2 above)
Comment (marcosrdz) [reply scripts/react-native-camera-kit.patch:1]: Its needed for v13. V14 is still beta

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: android/build.gradle (hunk 4 above)
Comment (marcosrdz) [reply android/build.gradle]: My bad. That was during the trial and error process

File: screen/wallets/pleaseBackup.tsx (hunk 3 above)
Comment (marcosrdz) [reply screen/wallets/pleaseBackup.tsx:38]: Nav 6 deprecated the "dangerously" part

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2hz2e46cyc57u5yvlyv3gkwoza?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11851)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/kfqysktykfmnnhygvrcnaixtvy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11852)

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/sdh5rusr3suk7btzulasssumf4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11855)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wq3kgjgt7st4jqqp4zna4w5fne?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11859)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]

**Pull request merged**